src/providers/ethereum/EthereumLedgerProvider.js

```diff
@@ -14,4 +17,63 @@ export default class EthereumLedgerProvider extends LedgerProvider {
     const hex = Buffer.from(message).toString('hex')
     return app.signPersonalMessage(address.derivationPath, hex)
   }
+
+  async getAddresses () { // TODO: Retrieve given num addresses?
+    const app = await this.getApp()
+    const path = this._baseDerivationPath + '0/0'
+    const address = await app.getAddress(path)
+    return [
+      {
+        address: address.address,
+        derivationPath: path,
+        publicKey: address.publicKey
+      }
+    ]
+  }
+
+  async getUnusedAddress () {
+    const addresses = await this.getAddresses()
+    return addresses[0]
+  }
+
+  async getUsedAddresses () {
+    return this.getAddresses()
+  }
+
+  async sendTransaction (to, value, data) {
+    const app = await this.getApp()
+    const addresses = await this.getAddresses()
+    const address = addresses[0]
+    const from = address.address
+    const path = address.derivationPath
+
+    const nonce = await this.getMethod('getTransactionCount')(ensureHexStandardFormat(from))
+    const gasLimit = 300000 // TODO: Implement gas estimation - this is safe for now
```

Comment:
We can use this: https://github.com/ethereum/wiki/wiki/JSON-RPC#eth_estimategas

Reply:
Thanks, yes ideally use this. atm, feel it would take a bit of playing about to get right. Since it will be negligible difference for the current swap usecase, prefer to come back to this once there is usage.

Reply:
have created this issue for tracking https://github.com/liquality/chainabstractionlayer/issues/161